raiden/channel.py

```diff
@@ -1135,20 +1116,18 @@ def create_mediatedtransfer(self, transfer_initiator, transfer_target, fee,
         )
         return mediated_transfer
 
-    def create_refundtransfer_for(self, transfer):
-        """ Return RefundTransfer for `transfer`. """
-        lock = transfer.lock
-
-        if not self.our_state.balance_proof.is_pending(lock.hashlock):
-            raise ValueError('Unknow hashlock')
+    def state_transition(self, state_change):
+        if isinstance(state_change, Block):
+            self.block_number = state_change.block_number
+            settlement_end = self.external_state.closed_block + self.settle_timeout
 
-        locked_transfer = self.create_lockedtransfer(
-            lock.amount,
-            1,  # TODO: Perhaps add identifier in the refund transfer too?
-            lock.expiration,
-            lock.hashlock,
-        )
+            if self.state == 'closed' and self.block_number > settlement_end:
+                self.external_state.settle()
 
-        cancel_transfer = locked_transfer.to_refundtransfer()
+        if isinstance(state_change, ContractReceiveClosed):
+            if state_change.channel_address == self.channel_address:
```

Comment:
the Block state change, used when a new block is mined, does not have a channel_address field